libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -67,16 +70,36 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     return this.config.flavor === 'drawer';
   }
 
+  willClose$ = from(this.willClose).pipe(first());
+  private navigationEndListener$: any = this.router.events.pipe(
+    takeUntil(this.willClose$),
+    filter((event) => event instanceof NavigationEnd)
+  );
+
   constructor(
     private injector: Injector,
     private elementRef: ElementRef<HTMLElement>,
     private renderer: Renderer2,
-    private resizeObserverService: ResizeObserverService
+    private resizeObserverService: ResizeObserverService,
+    private modalOutlet: ModalOutlet,
+    private router: Router
   ) {
     this.observeViewportResize();
   }
 
   ngOnInit(): void {
+    this.modalOutlet.activate(this);
```

Comment:
Consider supplying only relevant part of the dependency:
```suggestion
    this.modalOutlet.activate(this.config);
```